inc/field/fieldsfield.class.php

```diff
@@ -0,0 +1,527 @@
+use PluginFormcreatorAbstractField;
+use Dropdown;
+use DbUtils;
+use Plugin;
+use Ticket;
+use Change;
+use Problem;
+use Session;
+use Html;
+use OperatingSystem;
+use PluginFieldsDropdown;
+use PluginFieldsField;
+use PluginFormcreatorQuestion;
+use User;
+
+class FieldsField extends PluginFormcreatorAbstractField
+{
+
+   /** @var PluginFieldsField $field */
+   public $field = null;
+
+   /**
+    *
+    * @param array $question PluginFormcreatorQuestion instance
+    */
+   public function __construct(PluginFormcreatorQuestion $question) {
+      $this->question  = $question;
+
+      $decodedValues = json_decode($this->question->fields['values'], JSON_OBJECT_AS_ARRAY);
+      $field_name = $decodedValues['dropdown_fields_field'] ?? '';
+      $fieldObj = new PluginFieldsField();
+      if ($fieldObj->getFromDBByCrit(['name' => $field_name])) {
+         $this->field  = $fieldObj;
+      }
+   }
+
+   public function isPrerequisites(): bool {
+      return  (!(new Plugin())->isActivated('field'));
+   }
+
+   public static function getFieldsFromBlock($block_id): array {
+      global $DB;
+      $optgroup = [];
+
+      // Get all fields for block (except header)
+      $request = [
+         'SELECT' => ['name', 'label'],
+         'FROM'   => PluginFieldsField::getTable(),
+         'WHERE'  => [
+            'AND' => [
+               'plugin_fields_containers_id'      => $block_id,
+               'is_active' => true,
+               ['NOT' => ['type' => 'header']]
+            ]
+         ]
+      ];
+
+      $iterator = $DB->request($request);
+      foreach ($iterator as $row) {
+         $optgroup[$row['name']] = $row['label'];
+      }
+
+      return $optgroup;
+   }
+
+   public function getBlocks() {
+      global $DB;
+      $optgroup = [];
+
+      // Get all block for :
+      // Ticket / Change / Problem
+      // Activated
+      // Type -> dom
+      $dbUtils = new DbUtils();
+      $entityRestrict = $dbUtils->getEntitiesRestrictCriteria(PluginFieldsContainer::getTable(), "", "", true, false);
+      if (count($entityRestrict)) {
+         $entityRestrict = [$entityRestrict];
+      }
+
+      $request = [
+         'SELECT' => ['id', 'label'],
+         'FROM'   => PluginFieldsContainer::getTable(),
+         'WHERE'  => [
+            'AND' => [
+               'OR' => [
+                  ['itemtypes' => ['LIKE', '%\"'.Ticket::getType().'\"%']],
+                  ['itemtypes' => ['LIKE', '%\"'.Change::getType().'\"%']],
+                  ['itemtypes' => ['LIKE', '%\"'.Problem::getType().'\"%']],
+               ],
+               'type'      => 'dom',
+               'is_active' => true,
+               ]
+               + $entityRestrict,
+         ]
+      ];
+
+      $iterator = $DB->request($request);
+      foreach ($iterator as $row) {
+         $optgroup[$row['id']] = $row['label'];
+      }
+
+      return $optgroup;
+   }
+
+   public function getDesignSpecializationField(): array {
```

Comment:
Note : This method will be replaces by showForm, in the interface of this class.

you will have much less constraints. Lets see if this PR is merged before or after my task.

Reply:
refactor done